### PR TITLE
Add the bare mode to VAST

### PR DESCRIPTION
**Pull request by dominiklohmann**
### :notebook_with_decorative_cover: Description

This PR solves a multitude of issues we've had in one fell swoop:

- Data, configuration and library dirs interpreted relative to the VAST binary
  file are no longer specified as fixed paths. E.g., we now support
  running integration tests and relocateable binaries on systems where the
  libary install directory is `lib64` instead of `lib`.

- Unless the new `--bare-mode` (supersedes
  `--disable-default-config-directories`) is enabled, non-relocateeable VAST
  binaries no longer look for configuration, schemas, and plugins in directories
  relative to the binary location.

- Relocateable VAST binaries no longer look for configuration, schemas, and
  plugins in their original install directory, and instead always use paths
  relative to their binary location.

- With `--bare-mdoe` enabled, VAST now unload static plugins that are not
  explicitly specified in an explicitly specified configuration file. This
  allows for running VAST's test suites for static VAST binaries.

### :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

FIle-by-file. Test carefully.

The plugin loading change can be tested by building the static binary with the PCAP plugin enabled, which fails the `vast dump` integration test without this change.

Note that this change is not of very high priority, so I don't mind this sitting a bit until we have time to test it carefully.

### CODE REVIEW

File: libvast/src/detail/installdirs.cpp [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  return binary->parent_path().parent_path();
+}
+
+} // namespace
+
+std::filesystem::path install_datadir([[maybe_unused]] bool bare_mode) {
+#if VAST_ENABLE_RELOCATABLE_INSTALLATIONS
+  return install_prefix() / VAST_DATADIR;
+#else
+  if (!bare_mode)
```
Comment (dominiklohmann) [inline libvast/src/detail/installdirs.cpp:31]:
```suggestion
  if (bare_mode)
```

Comment (dominiklohmann) [reply libvast/src/detail/installdirs.cpp:31]: TODO: Remove the bare mode fallback, change integration tests for non-relocatable builds to use VAST from CMAKE_INSTALL_FULL_BINDIR.

File: libvast/vast/config.hpp.in [hunk 2]
```diff
@@ -32,9 +32,19 @@ extern const char* build_tree_hash;
...
+
+#if VAST_ENABLE_RELOCATABLE_INSTALLATIONS
+#  define VAST_DATADIR "@CMAKE_INSTALL_DATADIR@/vast"
+#  define VAST_CONFIGDIR "@CMAKE_INSTALL_SYSCONFDIR@/vast"
+#  define VAST_PLUGINDIR "@CMAKE_INSTALL_LIBDIR@/vast/plugins"
+#else
+#  define VAST_FULL_DATADIR "@CMAKE_INSTALL_FULL_DATADIR@/vast"
+#  define VAST_FULL_CONFIGDIR "@CMAKE_INSTALL_FULL_SYSCONFDIR@/vast"
+#  define VAST_FULL_PLUGINDIR "@CMAKE_INSTALL_FULL_LIBDIR@/vast/plugins"
+#endif
```
Comment (tobim) [inline libvast/vast/config.hpp.in:47]: To avoid accidental misuse because of autocompletion or something similar, I think we should move these and the implementation of the `installdirs` functions into `config.cpp.in`.

File: vast/vast.cpp [hunk 3]
```diff
@@ -55,13 +55,13 @@ int main(int argc, char** argv) {
...
 #ifdef VAST_ENABLED_PLUGINS
   // If plugins are enabled at compile time to always be loaded, add them here
   // if they were not already configured as part of vast.plugins, unless
-  // --disable-default-config-dirs is set, which would break the integration
-  // tests that rely on this feature to avoid accidentally loading undesired
-  // plugins.
-  if (!caf::get_or(cfg, "vast.disable-default-config-dirs", false))
+  // --bare-mode is set, which would break the integration tests that rely on
+  // this feature to avoid accidentally loading undesired plugins.
+  if (!bare_mode)
```
Comment (tobim) [inline vast/vast.cpp:64]: The entire comment is quite hard to follow. how about:
```
// Add plugins that were configured to be auto loaded at compile time.
// --bare-mode disables implicit plugins to simplify integration testing.
```

**Review by tobim — CHANGES_REQUESTED** [review]
I like the `installdirs` abstraction :+1:
Just two small things that I think can be improved.

File: libvast/vast/config.hpp.in (hunk 2 above)
Comment (dominiklohmann) [reply libvast/vast/config.hpp.in:47]: I opted to configure the `installdirs.cpp` file instead to avoid having to include `<filesystem>` in `config.hpp`.

**Review by tobim — APPROVED** [review]
